protostar/starknet_gateway/gateway_facade.py

```diff
@@ -203,12 +206,66 @@ async def call(
         register_response({"result": str(result._asdict())})
         return result
 
+    async def invoke(
+        self,
+        contract_address: int,
+        function_name: str,
+        account_address: str,
+        signer: BaseSigner,
+        inputs: Optional[CairoOrPythonData] = None,
+        max_fee: Optional[int] = None,
+        auto_estimate_fee: bool = False,
+    ) -> InvokeResult:
+        register_response = self._register_request(
+            action="INVOKE",
+            payload={
+                "contract_address": contract_address,
+                "function_name": function_name,
+                "max_fee": max_fee,
+                "auto_estimate_fee": auto_estimate_fee,
+                "inputs": str(inputs),
+                "signer": str(signer),
+            },
+        )
+
+        contract_function = await self._create_contract_function(
+            contract_address,
+            function_name,
+            client=AccountClient(
+                address=account_address,
+                client=self._gateway_client,
+                signer=signer,
+            ),
+        )
+        result = await self._invoke_function(
```

Comment:
With the merging of #664. It should be wrapped in a try-except block.

Reply:
done